e2etest/bgp_test.go

```diff
@@ -74,7 +74,7 @@ var _ = ginkgo.Describe("BGP", func() {
 
 		hostport := net.JoinHostPort(ingressIP, port)
 		address := fmt.Sprintf("http://%s/", hostport)
-		_, err = runCommand(true, "wget", "-O-", "-q", address, "-T", "60")
+		_, err = runCommand(true, "wget", "-O-", "-q", address, "-T", "60", "-t", "40")
```

Comment:
I see the default for `-t` / `--tries` is 20.  I wonder if increasing this is what's needed, or if maybe adding `-w` / `--wait` ?  It's not super clear to me what the wait behavior is by default if a connection fails.  If it's really retrying 20 times and still not working, either we should wait longer between retries, or maybe it's a real failure.

Reply:
```
‘--waitretry=seconds’
If you don’t want Wget to wait between every retrieval, but only between retries of failed downloads, you can use this option. Wget will use linear backoff, waiting 1 second after the first failure on a given file, then waiting 2 seconds after the second failure on that file, up to the maximum number of seconds you specify.

By default, Wget will assume a value of 10 seconds.
```

10 seconds is more the enough

Reply:
Is there a link to the failed CI job? I went into circleci at https://app.circleci.com/pipelines/github/metallb/metallb?status=failed and I dont see it anymore (it failed in #882). Maybe they get purged after the PR is merged?

I think the `--waitretry` options seems more specific.

Reply:
From what I remember on the failure in #822, it failed when wget exited non-zero (code 4, or "Network failure"), and didn't spot anything else obvious.  It also didn't collect all the logs from the cluster, but maybe that run was from before we merged the PR to collect logs ...

Reply:
It seems like `waitretry` should help and wont hurt. If it happens again, we can look through the logs again. It also looks like `connection refused` or `404` will not cause a retry . Maybe we could also add `--retry-connrefused`.

Reply:
@russellb Is this the same race condition you saw a few weeks ago?

Reply:
I don't think it's what I saw when running tests a few weeks ago.

I just did a local test.  I ran `inv dev-env -p layer2` and then ran `wget http://172.18.0.6 -T 60`, which is an IP in the MetalLB config, but is not allocated to any Service.  The result is it fails with `failed: No route to host` and an exit code of 4.  It does *not* retry.  I'm guessing this is the scenario we saw happen in CI, where we tried to connect before the network was ready.

Adding `-t` does not make it retry, nor does `--retry-connrefused`.  I'm guessing wget treats this as a fatal error where retrying is pointless.  Maybe we need to add retrying into the test code if the failure code is 4 (Network failure) ?

Reply:
Ok. I think the bash loop might be the best way to do it. @msherif1234 Do you need help or will you do it?

Reply:
bash loop or go loop?

Reply:
:D sorry. This morning I have been looking at Go, Python, Bash and C so I am all mixed up. Go would be better here, of course :D

Reply:
I add retry loop for wget PTAL